app/models/mixins/process_tasks_mixin.rb

```diff
@@ -51,6 +49,51 @@ def invoke_tasks_local(options)
       end
     end
 
+    def invoke_tasks_remote(options)
+      ApplicationRecord.group_ids_by_region(options[:ids]).each do |region, ids|
+        remote_options = options.merge(:ids => ids)
+        hostname = MiqRegion.find_by_region(region).remote_ws_address
+        if hostname.nil?
+          $log.error("An error occurred while invoking remote tasks...The remote region [#{region}] does not have a web service address.")
+          next
+        end
+
+        begin
+          invoke_api_tasks(hostname, remote_options)
+        rescue NotImplementedError => err
+          $log.error("#{base_class.name} is not currently able to invoke tasks for remote regions")
+          $log.log_backtrace(err)
+          next
+        rescue => err
+          # Handle specific error case, until we can figure out how it occurs
+          if err.class == ArgumentError && err.message == "cannot interpret as DNS name: nil"
+            $log.error("An error occurred while invoking remote tasks...")
+            $log.log_backtrace(err)
+            next
+          end
+
+          $log.error("An error occurred while invoking remote tasks...Requeueing for 1 minute from now.")
+          $log.log_backtrace(err)
+          MiqQueue.put(
```

Comment:
We'll need to add some more logic for this later so that we can prevent this from bouncing around the queue forever in the case of an unrecoverable error. Also, if a significant amount of time passes before the original request can be successfully service we would want to stop  trying. We already have a [PT story](https://www.pivotaltracker.com/story/show/124179701) for that. I just wanted to make note of it.